### ISSUE TITLE
Do not rename SideCondition variables when applying equations

### ISSUE THREAD

**Issue by ttuegel** [kframework/kore#1815]
`checkRequires` renames the variables of the `SideCondition` before sending its query to the solver. Instead, we should rename the variables in the result _before_ calling `checkRequires`: if the check passes, we will need the renamed equation anyway, and even if the check does not pass, the equation's result is almost always smaller than the `SideCondition` and faster to rename.

**Comment by ttuegel** [comment]
I am electing to postpone this after #1545.

**Comment by ttuegel** [comment]
This is already half-way done: The `requires` is already renamed. Next, we need to change every function in `Kore.Equation.Application` which takes a `SideCondition (Target variable)` argument to take a `SideCondition variable` argument instead. These functions are:

- [ ] `attemptEquation`
- [ ] `checkRequires`
- [ ] `targetEquationVariables`
